### PR TITLE
Feature/oauth migrations

### PR DESCRIPTION
**Pull request by MWL91**

### CODE REVIEW

File: database/migrations/2016_06_01_000001_create_oauth_auth_codes_table.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        if (Schema::hasTable('oauth_auth_codes')) return;
+
+        $this->schema->create('oauth_auth_codes', function (Blueprint $table) {
+            $table->string('id', 100)->primary();
+            $table->unsignedBigInteger('user_id')->index();
+            $table->uuid('client_id');
```
Comment (qunabu) [inline database/migrations/2016_06_01_000001_create_oauth_auth_codes_table.php:38]: this shouldn't be `index` ?

File: database/migrations/2016_06_01_000002_create_oauth_access_tokens_table.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        if (Schema::hasTable('oauth_access_tokens')) return;
+
+        $this->schema->create('oauth_access_tokens', function (Blueprint $table) {
+            $table->string('id', 100)->primary();
+            $table->unsignedBigInteger('user_id')->nullable()->index();
```
Comment (qunabu) [inline database/migrations/2016_06_01_000002_create_oauth_access_tokens_table.php:37]: those shouldn't been deleted when user is delete like 

```php
$table->foreign('user_id')
->references('id')->on('users')
->onDelete('cascade'); 
```

File: database/migrations/2016_06_01_000001_create_oauth_auth_codes_table.php (hunk 1 above)
Comment (MWL91) [reply database/migrations/2016_06_01_000001_create_oauth_auth_codes_table.php:38]: That's are standard tables from passport. It uses default setting, and i wouldn't like to change that.

Comment (qunabu) [reply database/migrations/2016_06_01_000001_create_oauth_auth_codes_table.php:38]: I think its pointless to put those files in our package 

You can run those migration from packages 

`php artisan migrate --package=vendor/package`

and add this to documentation that you can publish 

[Official documentation](https://laravel.com/docs/master/passport#installation) says you don't need doing anything at all 

image


* https://stackoverflow.com/questions/16818240/how-to-automatically-run-the-migrations-in-a-laravel-package
* https://laravel.com/docs/4.2/migrations#running-migrations

**Comment by qunabu** [comment]
up up

**Review by pa-cholek — APPROVED** [review]

**Pull request closed without merging**